lib/kazan/watcher.ex

```diff
@@ -0,0 +1,164 @@
+defmodule Kazan.Watcher do
+  @moduledoc """
+
+  Watches for changes on a resource.  This works by creating a HTTPoison ASync request.
+  The request will eventually timeout, however the Watcher handles recreating the request
+  when this occurs and requesting new events that have occurred since the last *resource_version*
+  received.
+
+  To use:
+
+  1. Create a request in the normal way, but setting `watch=true` e.g.
+  ```
+  request = Kazan.Apis.CoreV1.list_namespace!(watch: "true")
+  ```
+  2. Start a watcher using the request, and passing the initial resource version
+  and a pid to which to send received messages.  The initial *resource_version* can be
+  obtained by doing a normal, non-watch request to the same endpoint.
+  ```
+  Kazan.Watcher.start_link(request, resource_version: rv, send_to: self())
+  ```
+  3. In your client code you receive messages for each `%Versioned.Event{}`.
+    You can pattern match on the object type if you have multiple watchers configured.
+    For example, if the client is a `GenServer`
+  ```
+    # type is "ADDED", "DELETED" or "MODIFIED"
```

Comment:
I'm wondering if we should convert these strings into atoms to make this slightly more idiomatic....

Reply:
If we do that then we are no longer meeting the specification of `Kazan.Models.Apimachinery.Meta.V1.WatchEvent` which says `type` is String.

Reply:
Hmm...  Yeah, I suppose that's fair enough, lets leave for now...